### DISCUSSION TITLE
How can I set headless mode?

### DISCUSSION THREAD

**Discussion by devrsi0n** [Milkdown/milkdown#107] [Q&A] [answered]
Hi folks, I noticed there is a `headless` mode which is what I want, but I can't find any document explain how to enable that configuration.
Should we polish the doc a bit? Thanks in advance.
![image](https://user-images.githubusercontent.com/7880675/130324136-e5fedc8b-070e-4ea3-9d6a-c34e34663d7c.png)

**Comment by Saul-Mirone** [comment, marked as answer]
For now, you can set it like this:
```typescript
import { commonmarkNodes, commonmarkPlugins, heading, paragraph } from '@milkdown/preset-commonmark';

const nodes = commonmark
    .configure(paragraph, {
        headless: true
    })
    .configure(heading, {
        headless: true
    });

new Editor().use(nodes).use(commonmarkPlugins);
```

There is a feature that is not released yet, that you can simple set all nodes to headless like:
```typescript
const nodes = commonmark.headless();

new Editor().use(nodes).use(commonmarkPlugins);
```

I still need to make sure all plugins and nodes support headless mode before ship this feature, but it won't take long time.

**Reply by devrsi0n** [reply]
Thanks, but it seems we should use commonmarkNodes.
```ts
const nodes = commonmarkNodes
  .configure(paragraph, {
    headless: true,
  })
  .configure(heading, {
    headless: true,
  });
```
`commonmark` doesn't work.

**Reply by Saul-Mirone** [reply]
Yeah, only commonmarkNodes, gfmNodes and tableNodes work for now, but I plan to support to use it directly on commonmark and gfm.

**Reply by Saul-Mirone** [reply]
Update: these preset and plugins supports headless mode now, will be released in next version.

- [x] Commonmark
- [x] Table
- [x] Slash
- [x] Tooltip
- [x] GFM
- [x] Emoji
